Admin/assets/firebase/all-bills.js

```diff
@@ -13,48 +13,87 @@ firebase.auth().onAuthStateChanged((user) => {
-                                    <span class="legend-indicator bg-secondary"></span>Not Generated
-                                  </span>`
-                }
-                else if (billStatus==1){
-                    statusLine = `<span class="badge badge-soft-warning">
-                                    <span class="legend-indicator bg-warning"></span>Pending
-                                  </span>`
-                }
-                else {
-                    statusLine = `<span class="badge badge-soft-success">
-                                    <span class="legend-indicator bg-success"></span>Paid
-                                  </span>`
-                }
-                let invoice = `<a class="btn btn-sm btn-white" href="invoice.html?storeUID=${storeUID}&weekID=${weekNum}">
-                <i class="tio-receipt-outlined mr-1"></i> Invoice</a>`
-
-                var temp = [storeName,startDate,endDate,statusLine,total,invoice]
-                dataSet.unshift(temp)
-                }
-            })
+      jQuery.each(details.payment.weeks, function (weekNum, weekDetails) {
+        if (weekDetails && weekNum != "start") {
+          let storeData = {
+            "storeName": details.details.storeName,
+            "billStatus": weekDetails.billStatus, // Not checking if data is undefined or not, as It will be checked later
+            "weeklySales": weekDetails.sales
+          };
+          if (sortedByWeeks[weekNum]) {
+            sortedByWeeks[weekNum][storeUID] = storeData;
+          } else {
+            sortedByWeeks[weekNum] = {
+              [storeUID]: storeData
+            };
+          }
         }
-    })
-    updateDataTable(dataSet)
+      })
+    }
+  })
+
+  let dataSet = []
+  jQuery.each(sortedByWeeks, function (weekNum, allStores) {
+    if (allStores) {
+      jQuery.each(allStores, function (storeUID, storeData) {
+        let storeName = storeData.storeName ? storeData.storeName : 'No Store Name';
+        let startDate = getDateFromWeek(weekNum, 0);
+        let endDate = getDateFromWeek(weekNum, 1);
+        let billStatus = storeData.billStatus ? storeData.billStatus : -1;
+        let total = storeData.weeklySales ? storeData.weeklySales : 'N/A';
+        var statusLine
+        let mark = ``
+        if (billStatus == 0) {
+          statusLine = `<span class="badge badge-soft-secondary">
+                                <span class="legend-indicator bg-secondary"></span>Not Generated
+                              </span>`
+        } else if (billStatus == 1) {
+          statusLine = `<span id="${storeUID}${weekNum}Status" class="badge badge-soft-warning">
+                                <span class="legend-indicator bg-warning"></span>Pending
+                              </span>`
+
+          mark = `<button id="${storeUID}${weekNum}" class="btn btn-sm btn-white" onclick="updateBillStatus('${storeUID}','${weekNum}',this.id)">
```

Comment:
As I understand, button's ID can be obtained by `storeUID` and `weekNum`. 
1) If we just pass these two values to the function, we can obtain the button ID
2) Or if we just pass the button Id, we can obtain these values

so I think there's a redundancy here. Can we skip either of the params while calling the function?
Since we have to define the button's ID anyways, so we can just pass that and skip the first two.

To obtain individual values from the Id, we can use some separator like `-` or `_` between them in the Id

Reply:
Solved. 
1. Removed both( storeUID & weekNum ) params & passed only btn-id while calling function 
2. StoreUID & weekNum will be retrieved from btn-id by split function